internal/pkg/gateway/api.go

```diff
@@ -258,11 +258,55 @@ func (gs *Server) CommitStatus(ctx context.Context, signedRequest *gp.SignedComm
 	}
 
 	response := &gp.CommitStatusResponse{
-		Result: txStatus,
+		Result:      txStatus.Code,
+		BlockNumber: txStatus.BlockNumber,
 	}
 	return response, nil
 }
 
+// ChaincodeEvents supplies a stream of responses, each containing all the events emitted by the requested chaincode
+// for a specific block. The streamed responses are ordered by ascending block number. Responses are only returned for
+// blocks that contain the requested events, while blocks not containing any of the requested events are skipped. The
+// events within each response message are presented in the same order that the transactions that emitted them appear
+// within the block.
+func (gs *Server) ChaincodeEvents(signedRequest *gp.SignedChaincodeEventsRequest, stream gp.Gateway_ChaincodeEventsServer) error {
+	if signedRequest == nil {
+		return status.Error(codes.InvalidArgument, "a chaincode events request is required")
+	}
+
+	request := &gp.ChaincodeEventsRequest{}
+	if err := proto.Unmarshal(signedRequest.Request, request); err != nil {
+		return status.Error(codes.InvalidArgument, "invalid chaincode events request")
+	}
```

Comment:
How will an empty string or empty slice cause a nil-pointer panic? Could you give me an example of the change you are requiring here?

Reply:
In downstream code this byte array is used to unmarshal and I am not certain that every place all the checks have been in place. So, it was just to fail early if this byte slice is nil - kind of basic input check.

Reply:
You can unmarshall a zero-length `[]byte` fine. I don't understand the benefit of adding code bloat for a special check for a zero-length slice rather than letting a meaningful error be raised, such as the requested channel doesn't exist or the supplied identity is not valid. The client APIs don't send empty protobufs so it's not a scenario that can occur by accident